### PR TITLE
Created reseed file.  Updated with pipeline.

### PR DESCRIPTION
**Pull request by dwmorris11**

### CODE REVIEW

File: database/Reseed/index.js [hunk 1]
```diff
@@ -1,41 +1,28 @@
...
+    try {
+      await sequelize.authenticate();
+      console.log('Connection has been established successfully.');
+    } catch (error) {
+      console.error('Unable to connect to the database:', error);
+    }
+ })();
 
+ async findAttraction (id) => {
+  await const item = sequelize.findByPk(id);
```
Comment (JimKnape) [inline database/Reseed/index.js:26]: Item refers the attraction Id? A more informative variable name could be attraction or nearbyattraction?

File: database/Reseed/reseed.js [hunk 2]
```diff
@@ -1,18 +1,29 @@
...
 const attractionIds = [];
 
+let locations = [];
+
 const d = '|'; //delimiter
 
 const qty = 10000000; //quantity of records
 
-for (let i = 9000001; i <= qty; i += 1) {
+for (let i = 1; i <= qty; i += 1) {
```
Comment (JimKnape) [inline database/Reseed/reseed.js:26]: I am not sure if this has much of an impact, but creating an array of 10M records for the padded number requires you to iterate over the list of 10M twice (here and then in the pipe process) - could factor the building the padded id into the pipe stream and in line 106 forloop just declare a limit of 10M entries.

File: database/Reseed/reseed.js [hunk 3]
```diff
@@ -91,49 +102,111 @@ const makeNearbyExperience = function (attractionId) {
...
-  const address = String(chance.address({ short_suffix: true }));
-  const website = chance.email();
-  const phonenumber = chance.phone();
-  const email = chance.email();
-  const nearByRestaurants = [];
-  const nearByAttractions = [];
-  for (let i = 0; i < 3; i += 1) {
-    nearByRestaurants.push(makeNearbyRestaurant(i, location, attractionId));
-    nearByAttractions.push(makeNearbyAttraction(i, location, attractionId));
+async function * generate() {
```
Comment (JimKnape) [inline database/Reseed/reseed.js:105]: I really like this way of generation - I am going to try it for the Arrango! nice work.

File: database/Reseed/reseed.js [hunk 4]
```diff
@@ -91,49 +102,111 @@ const makeNearbyExperience = function (attractionId) {
...
+    for (let i = 0; i < 3; i+= 1) {
+      yield makeNearbyRestaurant(i, locations[j], attractionIds[j]);
+    }
+  }
+}
 
-  function write(file, data, cb) {
-    if (!file.write(data)) {
-      file.once('drain', cb);
+async function * generate2() {
```
Comment (JimKnape) [inline database/Reseed/reseed.js:132]: Could rename generate to a more informative function name.

File: database/Reseed/reseed.js [hunk 5]
```diff
@@ -91,49 +102,111 @@ const makeNearbyExperience = function (attractionId) {
...
+  for( let j = 0; j < attractionIds.length; j++){
+    for (let i = 0; i < 3; i+= 1) {
+      yield makeNearbyAttraction(i, locations[j], attractionIds[j]);
+    }
+  }
+}
+
+async function * generate3() {
```
Comment (JimKnape) [inline database/Reseed/reseed.js:140]: generate3 could be a more informative variable name.

File: database/Reseed/reseed.js [hunk 6]
```diff
@@ -91,49 +102,111 @@ const makeNearbyExperience = function (attractionId) {
...
+  nearRead = Readable.from(generate()),
+  nearFile,
+  (err) => {
+    if(err) {
+      console.error('Nearby Pipeline failed', err);
     } else {
-      process.nextTick(cb);
+      console.log('Nearby Pipeline succeeded');
+      const finishNear = -1 * (startTime - Date.now())/1000;
+      console.log('Finished Nearby in ', finishNear, ' seconds')
```
Comment (JimKnape) [inline database/Reseed/reseed.js:155]: Really smart to add console this result here. Good work!

**Comment by JimKnape** [comment]
Looks good Dustin! I really like the pipe stream method you implemented. My couple comments are mostly on variable names. 

I also mentioned that it looks like you are creating an array of padded Id's first to iterate over. You could try to refactor to generate the padded id within a single for loop than having two iterations to 10M. 

Merge! Good work!

**Comment by dwmorris11** [comment]
Thanks Jim.  I will put in a ticket to modify the variable names.

**Pull request merged**